File: CS6011/Day21/app/src/main/java/com/example/lab11helloandroid/MainActivity.java
package com.example.lab11helloandroid;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
   boolean firstclick_=true;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void handleButton(View view) {
        final String MsTag = "MainActivity.Dd";
        Log.d(MsTag, "button was pressed....");
        TextView tv = findViewById(R.id.outputInfoID);
        if (firstclick_) {
            tv.setText("Welcome to CS 6011");
            firstclick_ = false;
        } else {
            tv.setText("You clicked the button again");
        }
    }


}

File: CS6018/Lab2/app/src/main/java/com/example/lab2version2/MainActivity.kt
package com.example.lab2version2

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.lab2version2.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater)


        binding.fragmentContainerView.getFragment<ClickFragment>().setButtonFunction {
            val drawFragment = DrawFragment()
            val transaction = this.supportFragmentManager.beginTransaction()
            transaction.replace(R.id.fragmentContainerView, drawFragment, "draw_tag")
            transaction.addToBackStack(null)
            transaction.commit()
        }
        setContentView(binding.root)
    }
    }
